pkg/pushsync/pushsync.go

```diff
@@ -365,13 +395,23 @@ func (ps *PushSync) pushToClosest(ctx context.Context, ch swarm.Chunk, retryAllo
 			if r.err != nil && r.attempted {
 				ps.metrics.TotalFailedSendAttempts.Inc()
 
-				// if the node has warmed up AND no other closer peer has been tried
-				if time.Now().After(ps.warmupPeriod) && !ps.skipList.HasChunk(ch.Address()) && ps.topologyDriver.IsWithinDepth(ch.Address()) {
-					ps.skipList.Add(peer, ch.Address(), skipPeerExpiration)
+				// if we have received 1 more results than peers attempted, return
+				// this means all attempts including establishing storage have already failed
+				if results > attempts {
```

Comment:
not sure that would be advantageous, as that bool is set once to avoid multiple attempts of trying to establish storage, 
and using that would require signinprogress to be set to false to be able to exit based on it? also we dont obviously know if the result was received by the select statement already

intention was to keep track of the attempts and results, and as we can only get 1 result from 1 peer attempt, and one extra from the self establishing, if we ever received more results than attempts, that would mean all attempts concluded already, we can exit